src/Castle.Windsor/MicroKernel/SubSystems/Conversion/TypeNameConverter.cs

```diff
@@ -130,9 +134,10 @@ private Type GetType(string name)
 		private bool InitializeAppDomainAssemblies(bool forceLoad)
 		{
 			var anyAssemblyAdded = false;
+#if FEATURE_APPDOMAIN
 			if (forceLoad || assemblies.Count == 0)
 			{
-				var loadedAssemblies = ReflectionUtil.GetLoadedAssemblies();
```

Comment:
I am not sure we should have done this. It was quite trivial to scan for assemblies eagerly by scanning the base directory.

Please see: https://github.com/cryosharp/fortress/blob/master/src/Fortress.Windsor/Core/Internal/ReflectionUtil.cs#L119